substrate/frame/revive/src/tests.rs

```diff
@@ -1134,12 +1134,53 @@ fn delegate_call() {
 			builder::bare_instantiate(Code::Upload(caller_wasm))
 				.value(300_000)
 				.build_and_unwrap_contract();
-		// Only upload 'callee' code
-		assert_ok!(Contracts::upload_code(RuntimeOrigin::signed(ALICE), callee_wasm, 100_000,));
+
+		// Instantiate the 'callee'
+		let Contract { addr: callee_addr, .. } =
+			builder::bare_instantiate(Code::Upload(callee_wasm))
+				.value(100_000)
+				.build_and_unwrap_contract();
 
 		assert_ok!(builder::call(caller_addr)
 			.value(1337)
-			.data(callee_code_hash.as_ref().to_vec())
+			.data((callee_addr, 0u64, 0u64).encode())
+			.build());
+	});
+}
+
+#[test]
+fn delegate_call_with_limits() {
```

Comment:
Needs to test the failure case for not enough `deposit_limit`. I think this is not enforced properly right now (see other comment).